test/http-test.js

```diff
@@ -454,6 +454,60 @@ describe('HTTP', function() {
     assert.strictEqual(blocks.length, 10);
   });
 
+  // depends on the previous test to generate blocks
+  it('should fetch block header by height', async () => {
+    // fetch corresponding header and block
+    const height = 7;
+    const header = await nclient.get(`/header/${height}`);
+    assert.equal(header.height, height);
+
+    const properties = [
+      'hash', 'version', 'prevBlock',
+      'merkleRoot', 'time', 'bits',
+      'nonce', 'height', 'chainwork'
+    ];
+
+    for (const property of properties)
+      assert(property in header);
+
+    const block = await nclient.getBlock(height);
```

Comment:
It could be worth it to use ```nclient.get(`/block/${height}\`);``` here instead, as to be similar to the header call with ```await nclient.get(`/header/${height}`);```.

Reply:
I'm only using `nclient.get('/header/${height}')` here because there is no `nclient.getBlockHeader`. There is a PR to add it here: https://github.com/bcoin-org/bclient/pull/21/files

I feel like the raw get request should be turned into using the wrapper once its merged into `bclient`.

Reply:
That's a reason why I think it's better to use `nclient.get` in most cases. I think there is also case to have a minimal RPC client for testing, instead of depending on bclient. Meanwhile bclient does not have any tests.

Reply:
> I think there is also case to have a minimal RPC/HTTP client for testing, instead of depending on bclient. 

I get this perspective and its generally not the best practice to couple tests for different repos together. `bclient` is partially tested in `bcoin` (which is better than no tests at all). Since `bclient` is used in production and is already a dependency of `bcoin`, I feel like using it in tests is a net positive, unless you want `bcoin` as a `devDependency` of `bclient` so that a full test suite can be written there.